ibis/backends/duckdb/compiler.py

```diff
@@ -16,6 +21,46 @@ class DuckDBSQLExprTranslator(AlchemyExprTranslator):
     _has_reduction_filter_syntax = True
 
 
+@compiles(dt.UInt64, "duckdb")
+def compile_uint64(element, compiler, **kw):
+    return "UBIGINT"
+
+
+@compiles(dt.UInt32, "duckdb")
+def compile_uint32(element, compiler, **kw):
+    return "UINTEGER"
+
+
+@compiles(dt.UInt16, "duckdb")
+def compile_uint16(element, compiler, **kw):
+    return "USMALLINT"
+
+
+@compiles(dt.UInt8, "duckdb")
+def compile_uint8(element, compiler, **kw):
+    return "UTINYINT"
+
+
+@ibis_dt.dtype.register(duckdb_engine.Dialect, dt.UInt64)
+def dtype_uint64(_, satype, nullable=True):
+    return ibis_dt.UInt64(nullable=nullable)
+
+
+@ibis_dt.dtype.register(duckdb_engine.Dialect, dt.UInt32)
+def dtype_uint32(_, satype, nullable=True):
+    return ibis_dt.UInt32(nullable=nullable)
+
+
+@ibis_dt.dtype.register(duckdb_engine.Dialect, dt.UInt16)
+def dtype_uint16(_, satype, nullable=True):
+    return ibis_dt.UInt16(nullable=nullable)
+
+
+@ibis_dt.dtype.register(duckdb_engine.Dialect, dt.UInt8)
+def dtype_uint8(_, satype, nullable=True):
+    return ibis_dt.UInt8(nullable=nullable)
```

Comment:
This implements the reverse mapping, turning the sqlalchemy dtypes back into an ibis schema.

If we keep a generic set of unsigned sqlalchemy types as defined here currently, the `dtype.register` calls here could actually be moved into `ibis.backends.base.sql.alchemy.datatypes` and generic for all dialects, since the mapping here would be consistent for all supporting backends. No need to specialize on duckdb here.

Reply:
That seems like a good idea.